<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="weights_name"    default="yolov5s_person" />
    <arg name="data_name"       default="door_handle" />

    <!-- Camera topic and weights, config arguments -->
    <arg name="image_topic"     default="/camera/color/image_raw" />
    <arg name="weights"         default="$(find yolov5_ros)/weights/$(arg weights_name).pt" />
    <arg name="data"            default="$(find yolov5_ros)/data/$(arg data_name).yaml" />
    <arg name="width"           default="640" />
    <arg name="height"          default="480" />
    <arg name="conf_thres"      default="0.25" />

    <!-- Node -->
    <node name="detector" pkg="yolov5_ros" type="detect.py" output="screen" respawn="true">
        <param name="image_topic"   value="$(arg image_topic)" />
        <param name="weights"       value="$(arg weights)" />
        <param name="data"          value="$(arg data)" />
        <param name="width"         value="$(arg width)" />
        <param name="height"        value="$(arg height)" />
        <param name="conf_thres"    value="$(arg conf_thres)" />
    </node>
</launch>
